checks/pinned_dependencies.go

```diff
@@ -33,23 +33,60 @@ const CheckPinnedDependencies = "Pinned-Dependencies"
 // We only declare the fields we need.
 // Github workflows format: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions
 type gitHubActionWorkflowConfig struct {
-	// nolint: govet
-	Jobs map[string]struct {
-		Name  string `yaml:"name"`
-		Steps []struct {
-			Name  string `yaml:"name"`
-			ID    string `yaml:"id"`
-			Uses  string `yaml:"uses"`
+	Jobs map[string]gitHubActionWorkflowJob
+	Name string `yaml:"name"`
+}
+
+// A Github Action Workflow Job.
+// We only declare the fields we need.
+// Github workflows format: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions
+type gitHubActionWorkflowJob struct {
+	Name     string                     `yaml:"name"`
+	Steps    []gitHubActionWorkflowStep `yaml:"steps"`
+	Defaults struct {
+		Run struct {
 			Shell string `yaml:"shell"`
-			Run   string `yaml:"run"`
+		} `yaml:"run"`
+	} `yaml:"defaults"`
+	RunsOn   stringOrSlice `yaml:"runs-on"`
+	Strategy struct {
+		Matrix struct {
+			Os []string `yaml:"os"`
+		} `yaml:"matrix"`
+	} `yaml:"strategy"`
+}
+
+// A Github Action Workflow Step.
+// We only declare the fields we need.
+// Github workflows format: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions
+type gitHubActionWorkflowStep struct {
+	Name  string `yaml:"name"`
+	ID    string `yaml:"id"`
+	Uses  string `yaml:"uses"`
+	Shell string `yaml:"shell"`
+	Run   string `yaml:"run"`
+	If    string `yaml:"if"`
+}
+
+// stringOrSlice is for fields that can be a single string or a slice of strings. If the field is a single string,
+// this value will be a slice with a single string item.
+type stringOrSlice []string
+
+func (s *stringOrSlice) UnmarshalYAML(value *yaml.Node) error {
+	var stringSlice []string
+	err := value.Decode(&stringSlice)
+	if err != nil {
```

Comment:
is it worth refining the error checking by looking at a specific error that indicate wrong type?

Reply:
I tested out changing the `runs-on` field to something that wasn't a string or a slice. The error was:
```
internal error: invalid yaml file: internal error: error decoding stringOrSlice Value: yaml: unmarshal errors: line 9: cannot unmarshal !!map into string
```
I did the same to a different field (`shell`), and it gave the same error (just not wrapped):
```
internal error: invalid yaml file: yaml: unmarshal errors: line 13: cannot unmarshal !!map into string
```
So I don't see a need to indicate a special error when `run-on` is the wrong type.